### PR TITLE
Added a fake DriveTrainVariant and a constant to disable the DriveTrain.

### PR DESCRIPTION
**Pull request by glowkate**
I would like a review on this pull request, if somebody is able to do so.
I will merge this in two days if nobody reviews it.

### CODE REVIEW

**Comment by Dangosaurus** [comment]
LGTM. It should probably be documented along the likes of DriveTrainVariant(#2) and LazySolenoid(#3). Not sure if that should be part of this PR or should be it's own issue.

**Review by Dangosaurus — COMMENTED** [review]
Oops I realized now I probably should have left a review comment.

File: src/main/java/ca/warp7/frc2022/subsystems/DriveTrain.java [hunk 1]
```diff
@@ -39,7 +41,12 @@ public static void setVariant(DriveTrainVariant variant) {
         if (driveTrainVariant != null) {
             throw new IllegalStateException("Cannot set drive train variant");
         }
-        driveTrainVariant = variant;
+        if (kEnableDriveTrain){
+            driveTrainVariant = variant;
+        }
+        else{
+            driveTrainVariant = new LazyDriveTrainVariant();
```
Comment (glowmouse) [inline src/main/java/ca/warp7/frc2022/subsystems/DriveTrain.java:48]: Consider pushing this up to the caller, which is [this line](https://github.com/Team865/FRC-2022/blob/830b0d62205316c37b8030691a3a07de521eec53/src/main/java/ca/warp7/frc2022/Robot.java#L44) in robotInit()

>         System.out.println("Hello me is robit!");
> 
>         DriveTrain.setVariant(new FalconDriveTrainVariant());
> 
>         scheduler = CommandScheduler.getInstance();

The problem that I see is that if we don't have the drivetrain hardware then I think the **new FalconDriveTrainVariant()** will trigger the [FalconDriverTrainVariant constructor,](https://github.com/Team865/FRC-2022/blob/830b0d62205316c37b8030691a3a07de521eec53/src/main/java/ca/warp7/frc2022/subsystems/drivetrain/FalconDriveTrainVariant.java#L10) which will try to connect to the non-existent motors and gear box solenoids.  My guess is that it'll generate a bunch of CAN bus warnings on the console output and it'll work fine, but that's a guess.

Another thought is to just make the driveTrainVariant in the constructor for DriveTrain() and remove the setVariant method completely.  driveTrainVariant becomes a non-static member of DriveTrain, but I think that's okay...  The only way to create a DriveTrain appears to be DriveTrain's getInstance method;  the DriveTrain constructor is private.  DriveTrain is a classic singleton.

File: src/main/java/ca/warp7/frc2022/subsystems/DriveTrain.java [hunk 2]
```diff
@@ -39,7 +41,12 @@ public static void setVariant(DriveTrainVariant variant) {
...
+        }
     }
 
     private final LazySolenoid shifterSolenoid =
```
Comment (glowmouse) [inline src/main/java/ca/warp7/frc2022/subsystems/DriveTrain.java:52]: The LazySolenoid here also needs a be tied into the kEnableDriveTrain

Also, are we missing something like a kEnableDriveTrainShifter constant?  i.e., should this be

`new LazySolenoid(kDriveShifterID, kEnableSolenoids && kEnableDriveTrain && kEnableDriveTrainShifter);`

**Pull request merged**